### DISCUSSION TITLE
Jax Grad for scipy coo_matrix

### DISCUSSION THREAD

**Discussion by ZeshengLiu22** [jax-ml/jax#13503] [Q&A]
Hi,

Now I am doing a matrix multiplication loss =  V.T @ M @ V, where V is a device array and M is a spicy coo_matrix returned from some other function that can not be modified. M is 65536*65536 so it is impossible to use .todense to convert it into NumPy array at first.

Is there some way to resolve this problem?

![image](https://user-images.githubusercontent.com/119650353/205425422-a69d3eea-74a0-42aa-a9ee-c397bcc79f21.png)

Code is this and here mat_laplacian is the coo_matrix with size at least 65536*65536.

![image](https://user-images.githubusercontent.com/119650353/205425390-ad19f764-5c2c-46bb-8faa-a997ceb4e9e3.png)

Get such error if directly use grad()

**Comment by jakevdp** [comment]
Matrices of type `scipy.sparse.coo_matrix` are not compatible with JAX transformations like autodiff. But you can get most of the same functionality via `jax.experimental.sparse.BCOO`. Here's how you can convert a scipy COO matrix to a JAX BCOO matrix:
```python
from jax.experimental import sparse
mat_bcoo = sparse.BCOO.from_scipy_sparse(mat_scipy_coo)
```

**Reply by ZeshengLiu22** [reply]
Thanks a lot!
